### ISSUE TITLE
Design and implement `GetK8sResourcesForInstalledPackage`

### ISSUE THREAD

**Issue by ppbaena** [kubeapps/kubeapps#3403]
Currently when an app is installed on the cluster, our AppView gets the data about the related k8s resources from the k8s api server.

We want to replace that with an API call which allows getting the (relevant) resources for a particular installed package (only).

This will be a huge step forward in removing the requirement to expose the k8s api server to the dashboard.

**Comment by antgamdia** [comment]
In order to avoid a never-ending issue, let's move some of the remaining open items from https://github.com/kubeapps/kubeapps/issues/3340 to this issue:

- [ ]  https://github.com/kubeapps/kubeapps/issues/3403
- [ ]  Use `GetK8sResources` in the UI (blocked)
    - [ ] Remove old kubeops references for retrieving k8s resources?

_Updated on September 28th_

**Comment by absoludity** [comment]
After successfully testing out some ideas to enable a separate "resources" plugin with a single streaming endpoint for watching multiple/many resources, I've added a [section in the design doc for the GetK8sResources endpoint](https://docs.google.com/document/d/1z03msZRGsRvcRaom-yrvEwIWcEfNy6fSA5Zg28gjYvA/edit#heading=h.2pgw6nn9xing), as well as updating the core packaging API with an extra endpoint for [GetInstalledPackageResourceRefs](https://docs.google.com/document/d/1z03msZRGsRvcRaom-yrvEwIWcEfNy6fSA5Zg28gjYvA/edit#heading=h.5e95xks1mhfz) (which is specific for each packaging backend).

If you can take a look when you have a chance, that'd be great. (cc @gfichtenholt as I don't think you're subscribed to all issue email :) ).

**Comment by gfichtenholt** [comment]
I will take a close look when I get a chance. How soon do you need it? Are you ready to starting adding code for this and and just waiting for me to review? Right now I am have like 3 non-trivial things on my plate that needed to be done yesterday :-) But I will drop everything, as usual, to unblock you as soon as you need it

Just a couple of superficial comments based on cursory examination:
1) you need to update InstalledPackageDetail to fix the TODO
2) don't understand why you called a section at the end "Other plugins..." as if we'd have plug-ins other than flux or helm. Also, this statement "All plugins that implement the core.packages interface will include a GetInstalledPackageResourceRefs method, returning the list of resource references for the resources created by the installed package. Getting or watching those resources does not need to be implemented separately by each packaging plugin and is instead provided by a separate resources plugin" makes it sound like we are about to add some more plug-ins. Are we? Just confusing, that's all. Probably need to discuss in a meeting for me to understand
3) confused by "plugins.resources.v1alpha1". Why is there nothing in the core to aggregate like for packaging stuff?

**Comment by absoludity** [comment]
> But I will drop everything, as usual, to unblock you as soon as you need it

Thanks Greg, but no need to drop anything. I only CC'd you in case you're interested, but I can go through this with Antonio early next week too. Or if you're interested, we can chat through it on a call, but no problem if you're busy.

> 
> Just a couple of superficial comments based on cursory examination:
> 
>     1. you need to update InstalledPackageDetail to fix the TODO

Will do - just focused on documenting what I've investigated and am planning, I'll go through and cleanup the TODO tomorrow.

> 
>     2. don't understand why you called a section at the end "Other plugins..." as if we'd have plug-ins other than flux or helm.

Flux, helm (and kapp-controller) are packaging plugins, but the apis server has been designed so we can support other types of plugins too. As seems to be the case for this work.

> Also, this statement "All plugins that implement the core.packages interface will include a GetInstalledPackageResourceRefs method, returning the list of resource references for the resources created by the installed package. Getting or watching those resources does not need to be implemented separately by each packaging plugin and is instead provided by a separate resources plugin" makes it sound like we are about to add some more plug-ins.

Yes, the functionality of getting and watching kubernetes resources seems to be a good example of another plugin. We don't need (or want) each packaging plugin (helm, flux, etc.) to re-implement this functionality. I tested it and am able to add this functionality as a separate plugin which works nicely, and other uses of the apis-server (outside of kubeapps)  can choose whether they want to enable this plugin or not.

> Are we? Just confusing, that's all. Probably need to discuss in a meeting for me to understand

Yep, let's try to do that. I'll ping you on slack to see when you've time, but again, no problem if you're busy.
> 
>     3. confused by "plugins.resources.v1alpha1". Why is there nothing in the core to aggregate like for packaging stuff?

**Comment by absoludity** [comment]
Moving to Done as the design is now implemented, and opening a separate issue for the dashboard changes to use the new API.
